hardware/README.md

```diff
@@ -148,6 +148,11 @@ through an AXI master interface. In the SNAP configuration step the existence of
 # NVMe support
 For FPGA cards with NVMe flash attached, the SNAP framework supports the integration of up to two Flash drives. Via the SNAP configuration option 'Enable NVMe' the instantiation of the NVMe host controller together with the corresponding PCIe root complexes and the required AXI interfaces can be configured.
 
+Some HDL and HLS based examples are available. In all cases the NVME driver used requires the data to transit through the card SDRAM Please see the [NVMe main documentation page](../hardware/doc/NVMe.md) .
+- Automated hardware check (read write of arbitrary data) can be performed using action [snap/actions/hdl_example](../actions/hdl_example)
+- file transfers using fixed 4k blocks read writes can be performed with [snap/actions/hdl_nvme_example](../actions/hdl_nvme_example)
+- generic memcopy can be performed using [snap/actions/hls_nvme_memcopy](../actions/hls_nvme_memcopy)
+
```

Comment:
Very good points. What do you think about re-ordering the logical sequence for clarity - maybe like this:

The actions directory contains HDL and HLS based examples on how to use NVMe:
- [snap/actions/hdl_example](../actions/hdl_example) provides automated data checking (read write of arbitrary data)
- [snap/actions/hdl_nvme_example](../actions/hdl_nvme_example) shows file transfers using fixed 4k block reads and writes
- [snap/actions/hls_nvme_memcopy](../actions/hls_nvme_memcopy) implements a generic HLS memcopy function

Note the data itself doesn't flow through the NVMe host controller, but transits through the on-card SDRAM. Please see the [NVMe main documentation page](../hardware/doc/NVMe.md) for more details.

Reply:
very good.